api/java/org/openyolo/api/ui/ProviderPickerActivity.java

```diff
@@ -56,9 +61,28 @@
 
     private static final String EXTRA_PROVIDER_INTENTS = "providerIntents";
     private static final String EXTRA_TITLE_RES = "titleRes";
+    private static final String EXTRA_USER_CANCELED_RESULT = "userCanceledResult";
     private static final String LOG_TAG = "ProviderPicker";
 
-    private CredentialClient mClient;
+    private static final ActivityResult RETRIEVE_USER_CANCELED_RESULT =
+            ActivityResult.of(
+                    CredentialRetrieveResult.CODE_USER_CANCELED,
+                    CredentialRetrieveResult.USER_CANCELED.toResultDataIntent());
```

Comment:
I mean that

```
CredentialRetrieveResult.USER_CANCELED.getResultCode() == CredentialRetrieveResult.CODE_USER_CANCELED
```

However, you are right that since the different result types (retrieve / save / hint etc) don't share a common base type, there's no straightforward way to actually call getResultCode() without knowing the type first. So, let's leave this for now.